big_o/reports.py

```diff
@@ -0,0 +1,26 @@
+from big_o.complexities import ALL_CLASSES
+
+def big_o_report(best, others):
+    """ Creates a human-readable report of the output of the big_o function.
+
+    Input:
+    ------
+
+    best -- Object representing the complexity class that best fits
+            the measured execution times.
+    others -- A dictionary of fitted complexity classes to the residuals
+
+    Output:
+    -------
+
+    report -- A string describing the report of the big_o function
+
+    """
+    report = ""
+    report += 'Best : {!s:<60s} \n'.format(best)
+    for class_, residuals in others.items():
+        if type(class_) in ALL_CLASSES:
+            report += '{!s:<60s}    (res: {:.2G})\n'.format(class_, residuals)
+        else:
+            report += '{!s:<60s} {}\n' .format(class_ + ':', residuals)
```

Comment:
Why do you need to distinguish between complexity classes in ALL_CLASSES or not?

Reply:
This in needed as the big_o called with 'return_raw_data=True' also returns 'measures' and 'times' along with it, Which might cause the program to break if not handled.

Reply:
I'm sorry I don't understand, here `big_o` is not called. The two `if ... else` blocks simply format `class_` and `residuals` slightly differently

Reply:
I understand now. I wouldn't print them out.

The correct check is `if isinstance(class_, ComplexityClass)` , because we want to allow users to define their own subclasses, which might not always be in `ALL_CLASSES`.

Please to not print `times` and `measures`, which are potentially very long and not so interesting

Reply:
Got it.

Reply:
Removed time and measures off the final list